File: Algorithm_461_Hamming Distance.cpp
#include <iostream>
#include <cmath>
using namespace std;
string convertToBinary(int x)
{
	string res="";
	while (x!=0)
	{
		res=char(x%2+'0')+res;
		x/=2;
	}
	return res;
}
int hammingDistance(int x, int y)
{
	string s1=convertToBinary(x);
	string s2=convertToBinary(y);
	int l1=s1.length(),l2=s2.length(),count=0;
	if (l2>l1) return hammingDistance(y,x);
	char c1,c2;
	for (int i=0;i<l1;i++)
	{
		c1=s1[l1-1-i];
		c2=i<l2?s2[l2-1-i]:'0';
		if (c1!=c2) count++;
	}
	return count;
}
int main()
{
	cout<<hammingDistance(1,4);
	return 0;
}

File: Algorithm_461_Hamming Distance.java
// My Solution
// 思路：循环32次，每次将x和y的最低位通过和1做与运算的方式提取出来，并做异或运算

class Solution {
    public int hammingDistance(int x, int y) {
        int count = 0;
        for (int i = 0; i < 32; i++) {
            count += 1 & (x ^ y);
            x >>= 1;
            y >>= 1;
        }
        return count;
    }
}

// Tricky Solution
class TrickySolution {
    public int hammingDistance(int x, int y) {
        return Integer.bitCount(x ^ y);
    }
}